dialogue-target/src/main/java/com/palantir/dialogue/Request.java

```diff
@@ -197,26 +199,38 @@ public Request.Builder headerParams(Multimap<String, ? extends String> entries)
         }
 
         public Request.Builder putAllHeaderParams(String key, Iterable<String> values) {
+            Preconditions.checkArgumentNotNull(key, "Header name must not be null");
```

Comment:
Let's match conjure builder semantics where we `Preconditions.checkNotNull` individual elements (like this one), but we don't iterate through collections (like `putAllHeaderParams(Multimap<String, ? extends String> entries)` below)

Reply:
I am curious how your bug came about, I would have expected the recent nullaway rollout to catch that sort of thing before hitting prod. Perhaps there's a case we're missing?

Reply:
I still have no idea how my bug came about, which is part of why I want to make this change.
I'm okay trying just checking non-collections for now, though I doubt there are ever enough query/path/header parameters that this will actually be a notable performance hit.